rskj-core/src/main/java/co/rsk/validators/BlockUnclesValidationRule.java

```diff
@@ -75,12 +77,15 @@ public boolean isValid(Block block) {
 
         if (!unclesHash.equals(unclesListHash)) {
             logger.warn("Block's given Uncle Hash doesn't match: {} != {}", unclesHash, unclesListHash);
-            panicProcessor.panic("invaliduncle", String.format("Block's given Uncle Hash doesn't match: %s != %s", unclesHash, unclesListHash));
+            panicProcessor.panic("invaliduncle", String.format(
+                    "Block's given Uncle Hash doesn't match: %s != %s", unclesHash, unclesListHash));
             return false;
         }
         List<BlockHeader> uncles = block.getUncleList();
-        if (CollectionUtils.isNotEmpty(uncles) && !validateUncleList(block.getNumber(), uncles, FamilyUtils.getAncestors(blockStore, block, uncleGenerationLimit), FamilyUtils.getUsedUncles(blockStore, block, uncleGenerationLimit)))
-        {
+        if (CollectionUtils.isNotEmpty(uncles) &&
```

Comment:
We usually create one or more explanatory variables when an `if` statement gets too complicated.

Reply:
It is not complicated. If the list is not empty, all the rules are in validateUncleList ( if they don't check )